Scripts/Tracking/Collision/List/SortCollisionsByNearest.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace VRTK.Core.Tracking.Collision.List
+{
+    using UnityEngine;
+    using VRTK.Core.Extension;
+
+    /// <summary>
+    /// Sorts the given collision collection based on which collision <see cref="Transform"/> components are nearest to the specified <see cref="source"/>.
+    /// </summary>
+    public class SortCollisionsByNearest : MonoBehaviour
+    {
+        /// <summary>
+        /// The source <see cref="Transform"/> to determine the closest collision to.
+        /// </summary>
+        [Tooltip("The source Transform to determine the closest collision to.")]
+        public Transform source;
+
+        /// <summary>
+        /// Emitted when the collection is sorted.
+        /// </summary>
+        public ActiveCollisions.UnityEvent Sorted = new ActiveCollisions.UnityEvent();
+
+        protected ActiveCollisions.EventData sortedList = new ActiveCollisions.EventData();
+
+        /// <summary>
+        /// Sorts the given collision collection by the collisions that are nearest to the source <see cref="Transform"/>.
+        /// </summary>
+        /// <param name="originalList">The original collision collection.</param>
+        public virtual void DoSort(ActiveCollisions.EventData originalList)
+        {
+            Sort(originalList);
+        }
+
+        /// <summary>
+        /// Sorts the given collision collection by the collisions that are nearest to the source <see cref="Transform"/> and returns the sorted collection.
+        /// </summary>
+        /// <param name="originalList">The original collision collection.</param>
+        /// <returns>The sorted collision collection.</returns>
+        public virtual ActiveCollisions.EventData Sort(ActiveCollisions.EventData originalList)
+        {
+            if (!isActiveAndEnabled || source == null)
+            {
+                return originalList;
+            }
+
+            sortedList.Set(originalList);
+
+            sortedList.activeCollisions.Sort(
+            (data1, data2) =>
+                {
+                    float distance1 = Vector3.Distance(data1.collider.GetContainingTransform().position, source.position);
+                    float distance2 = Vector3.Distance(data2.collider.GetContainingTransform().position, source.position);
+                    int result = distance1.CompareTo(distance2);

```

Comment:
This looks strange as we don't introduce variables we are about to return as is if the expression is small enough. How about `return distance1.CompareTo(distance2);`?